creation/create_cvmfsexec_distros.sh

```diff
@@ -3,123 +3,196 @@
-    fi
-    echo "Skipping the building of cvmfsexec distribution tarballs..."
-    rm -rf "$cvmfsexec_latest"
-    exit 0
-else
-    # if current version and latest version are different
-    if [[ -z "$curr_ver" ]]; then
-        # $curr_ver is empty; first time run of factory upgrade
-        # no version info stored in work-dir/cvmfsexec/tarballs
-        echo "Building cvmfsexec distributions..."
-    else
-        # $curr_ver is not empty; subsequent run of factory upgrade (and not the first time)
-        echo "Found newer version of cvmfsexec..."
-        echo "Rebuilding cvmfsexec distributions using the latest version..."
-    fi
 
-    # build the distributions for cvmfsexec based on the source, os and platform combination
-    cvmfsexec_distros="$cvmfsexec_temp"/distros
-    if [[ ! -d "$cvmfsexec_distros" ]]; then
-        mkdir -p "$cvmfsexec_distros"
-    fi
+####################### MAIN SCRIPT STARTS FROM HERE #######################
 
-    declare -a cvmfs_sources
-    cvmfs_sources=($(echo $CVMFS_SOURCES | tr ":" "\n"))
-
-    declare -a machine_types
-    machine_types=($(echo $SUPPORTED_TYPES | tr ":" "\n"))
-
-    for cvmfs_src in "${cvmfs_sources[@]}"
-    do
-        for mach_type in "${machine_types[@]}"
-        do
-            echo -n "Making $cvmfs_src distribution for $mach_type machine..."
-            os=`echo $mach_type | awk -F'-' '{print $1}'`
-            arch=`echo $mach_type | awk -F'-' '{print $2}'`
-            if "$cvmfsexec_latest"/makedist -m $mach_type $cvmfs_src &> /dev/null ; then
-                "$cvmfsexec_latest"/makedist -o "$cvmfsexec_distros"/cvmfsexec-${cvmfs_src}-${os}-${arch} &> /dev/null
-                if [[ -e "$cvmfsexec_distros"/cvmfsexec-${cvmfs_src}-${os}-${arch} ]]; then
-                    echo " Success"
-                    tar -cvzf "$cvmfsexec_tarballs"/cvmfsexec_${cvmfs_src}_${os}_${arch}.tar.gz -C "$cvmfsexec_distros" cvmfsexec-${cvmfs_src}-${os}-${arch} &> /dev/null
-                fi
-            else
-                echo " Failed! REASON: $cvmfs_src may not yet have a $mach_type build."
-            fi
-
-            # delete the dist directory within cvmfsexec to download the cvmfs configuration
-            # and repositories for another machine type
-            rm -rf "$cvmfsexec_latest"/dist
-        done
-    done
-
-    # remove the distros and latest folder under cvmfsexec.tmp
-    rm -rf "$cvmfsexec_distros"
-    rm -rf "$cvmfsexec_latest"
+# parsing the command-line arguments
+# check whether the first argument passed is the option '--work-dir'
+if [[ $1 == "--work-dir" && ! -d "$2" ]]; then
+	# if value after this option is not a path (it is empty/unspecified), invoke the error handler
+	error_handler "The value of --work-dir option must be an existing directory."
+elif [[ -d "$1" ]]; then
+	# check if there was a path specified without the --work-dir option, invoke the error handler
+	error_handler "--work-dir option should precede directory path."
+fi
```

Comment:
This may cause some difficult error if there are directories named like the sources (osg, default, egi). I would skip this check and trust the following checks to catch a problem

Reply:
This condition is purely checking if a directory path was entered without the `--work-dir` option preceding the path value and not the name of the directory. So, its unclear to me as to how this would cause the kinds of error that you described. Maybe I am not understanding your comment correctly?

Reply:
Let's say I don't specify --work-dir (I want to use the default), so I call  `create_cvmfsexec_distros.sh osg` to build all platforms for osg, but I have locally a subdirectory called "osg". `[[ -d "$1" ]]` is true and I may not understand well what's happening because I forgot about that directory and I thought I was specifying a source.

Reply:
```
$0 [--work-dir DIR] SOURCES_LIST [PLATFORMS_LIST]
```
If there is no --work-dir, $1 is a SOURCES_LIST (first argument) that is verified right below. The error "invalid source $1" and the usage string printed will be clear enough to understand that --work-dir needs to be added

Reply:
Discussed on Slack and agreed to incorporate this suggestion. The check has been removed since the check for sources will cover the scenario that I described in my response to the original comment in this thread.